### PR TITLE
CMake: Add 3.29

### PR DESCRIPTION
**Pull request by johnwparent**

### CODE REVIEW

**Review by kwryankrattiger — DISMISSED** [review]
Confirmed checksum.

**Comment by alalazo** [comment]
https://gitlab.spack.io/spack/spack/-/jobs/10532859 says pod runner not found... restarting pipelines :sweat:

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by johnwparent** [comment]
@alalazo FWIW The GL failure looks legitimate, I think @danlipsa may have a patch for this but we're still investigating.
Thanks for trying to get it to pass!

**Comment by johnwparent** [comment]
cc @alalazo I've added the two most recent 3.29 patch releases as well

File: var/spack/repos/builtin/packages/cmake/package.py [hunk 1]
```diff
@@ -29,6 +29,8 @@ class Cmake(Package):
     license("BSD-3-Clause")
 
     version("master", branch="master")
+    version("3.29.2", sha256="36db4b6926aab741ba6e4b2ea2d99c9193222132308b4dc824d4123cb730352e")
+    version("3.29.0", sha256="a0669630aae7baa4a8228048bf30b622f9e9fd8ee8cedb941754e9e38686c778")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/cmake/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/proj/package.py [hunk 2]
```diff
@@ -89,6 +89,7 @@ class Proj(CMakePackage, AutotoolsPackage):
 
     patch("proj.cmakelists.5.0.patch", when="@5.0")
     patch("proj.cmakelists.5.1.patch", when="@5.1:5.2")
+    patch("proj-8.1-cmake-3.29-new-tiff-interface.patch", when="+tiff @:9.1.0 ^cmake@3.19:")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/proj/package.py]: Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

**Review by alalazo — CHANGES_REQUESTED** [review]
Basically LGTM. Only a couple of minor comments. 

Pipelines are still running, but if it is just `chai` and `raja` failing, I'll merge this (they are failing also on develop)

File: var/spack/repos/builtin/packages/cmake/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/cmake/package.py]: If we want to preserve 3.29.0, I think we should add it in the list of deprecated versions.

File: var/spack/repos/builtin/packages/proj/package.py (hunk 2 above)
Comment (johnwparent) [reply var/spack/repos/builtin/packages/proj/package.py]: It seems like the 3.28 PR is failing on this as well, I'm going to make this its own PR and apply suggested changes there. Will tag you (and remove this change here) once that's up.

**Comment by johnwparent** [comment]
Note: needs https://github.com/spack/spack/pull/43780

**Review by alalazo — APPROVED** [review]

**Pull request merged**